modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/Type.java

```diff
@@ -39,22 +42,82 @@
   boolean isAssignableFrom(Class<?> clazz);
 
   /**
-   * Checks the assignability of the current type to the given class
+   * Checks the assignability of the current type from the given {@link Type}
+   *
+   * @param type The type to check
+   * @return a boolean indicating whether the type is assignable or not to the given {@link Type}
+   * @since 4.1
+   */
+  boolean isAssignableFrom(Type type);
+
+  /**
+   * Checks the assignability of the current type to the given {@link Class}
    *
    * @param clazz The class to check
    * @return a boolean indicating whether the type is assignable or not to the given class
    * @since 4.1
    */
   boolean isAssignableTo(Class<?> clazz);
 
+  /**
+   * Checks the assignability of the current type to the given {@link Type}
+   *
+   * @param type The type to check
+   * @return a boolean indicating whether the type is assignable or not to the given {@link Type}
+   * @since 4.1
+   */
+  boolean isAssignableTo(Type type);
+
+  /**
+   * Checks equality of the current type to the given {@link Type}
+   *
+   * @param type The type to check equality
+   * @return a boolean indicating whether the type is the same type or not
+   * @since 4.1
+   */
+  boolean isSameType(Type type);
+
+  /**
+   * Checks equality of the current type to the given {@link Class}
+   *
+   * @param clazz The type to check equality
+   * @return a boolean indicating whether the type is the same type or not
+   * @since 4.1
+   */
+  boolean isSameType(Class<?> clazz);
+
+  /**
+   * @return a boolean indicating if the current type is instantiable or not.
+   * @since 4.1
+   */
+  boolean isInstantiable();
+
   /**
    * @return The generics for the current type.
    * @since 4.1
    */
   List<TypeGeneric> getGenerics();
 
-  //TODO: Remove once the type loader exists MULE-14040
-  java.lang.reflect.Type getReflectType();
+  /**
+   * @param interfaceClass The {@link Class} with generics
+   * @return The list of generics types from the given interface class.
+   * @since 4.1
+   */
+  List<Type> getInterfaceGenerics(Class interfaceClass);
+
+  /**
+   * @return The current type described as a {@link MetadataType}
+   * @since 4.1
+   */
+  MetadataType asMetadataType();
 
   String getTypeName();
+
+  /**
+   * @return The {@link ClassInformationAnnotation} describing the current {@link Type}
+   * @since 4.1
+   */
+  ClassInformationAnnotation getClassInformation();
+
+  boolean isAnyType();
```

Comment:
doesn't seem right to have this here. If we add this, then we could have many many more, let's remove it and let the consumer of this interface do `isAnyType(type.asMetadataType())`

Reply:
This is hack, I don't like it, but it's the only way, If I have a Type `Result<?,?>` In java is `Result<Object, Object>`, the `MetadataType` generated from those generics will be Object, Object, so to be able to don't break backwards the only way that I have is to check from the concrete type, this means the `java.lang.reflect.Type` or the `TypeMirror`.